nautobot/dcim/views.py

```diff
@@ -2537,7 +2536,7 @@ def queryset_to_csv(self):
             csv = csv_format(obj_data)
             csv_data.append(csv)
 
-        return "\n".join(csv_data)
+        return "\n".join([headers, *sorted(csv_data)])
```

Comment:
Was this the missing link??

Reply:
This was part of fixing the intermittent test failures; the other part was overriding the `test_list_objects_with_constrained_permission` test method to ensure that the remote end of the connection also had view permissions granted, which was needed to handle the intermittent case where:
1) the instance being arbitrarily chosen to be granted permissions and checked for was interface 1 (the test queryset is unsorted so there's no guarantee as to which of the test's interfaces are chosen), which happens to be connected to an interface on device 2
2) the view queryset ended up picking the pair (device2_interface, device1_interface1) instead of (device1_interface1, device2_interface)
3) therefore if we didn't *also* grant view permissions on device2_interface, this pair wouldn't be included in the queryset results

I think there's some further issues with this view and permissions (really we shouldn't present any connected pair unless the user has view permissions on both endpoints, but right now if I'm understanding things correctly it only actually cares about the first interface in the pair), but this is at least an incremental improvement.